### PR TITLE
fix: bump Jina version and Port ImageNormalizer and CLIPImageEncoder to use jinahub+docker

### PR DESCRIPTION
**Pull request by winstonww**
This PR bumps the Jina version of cross-modal-search to 2.0.4 and port ImageNormalizer and CLIPImageEncoder to use jinahub+docker

### CODE REVIEW

File: cross-modal-search/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
-jina[hub, http]==2.0.0rc5
+jina[hub, http]==2.0.4
```
Comment (mapleeit) [inline cross-modal-search/requirements.txt:1]: This better be `~=2.0.0`

Comment (numb3r3) [reply cross-modal-search/requirements.txt:1]:
```suggestion
jina[hub, http]~=2.0.0
```

Comment (numb3r3) [reply cross-modal-search/requirements.txt:1]: test

Comment (numb3r3) [reply cross-modal-search/requirements.txt:1]: test2

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: cross-modal-search/requirements.txt (hunk 1 above)
Comment (winstonww) [reply cross-modal-search/requirements.txt]: Addressed, please see [0e3acad](https://github.com/jina-ai/examples/pull/670/commits/0e3acad28fc311ec836f65e99209d5f1b84de393) for the change.

**Comment by winstonww** [comment]
The current PR depends on this [PR](https://github.com/jina-ai/executor-image-normalizer/pull/15). Need to pass `uint8` dtype to ImageNormalizer so that CLIPImageEncoder receives the expected input.

**Comment by nan-wang** [comment]
> The current PR depends on this PR. Need to pass uint8 dtype to ImageNormalizer so that CLIPImageEncoder receives the expected input.

You can convert it to `uint8` when converting image into `blob`. Check out the codes at 
https://github.com/jina-ai/eap-coohom/blob/main/pods/executors.py#L121
https://github.com/jina-ai/eap-coohom/blob/main/pods/executors.py#L24

**Comment by winstonww** [comment]
Thank you for the suggestion. I was wondering if it's possible to just pass `target_dtype` into `ImageNormalizer` with `numpy.uint8` as I believe this also works after the latest [commit](https://github.com/jina-ai/examples/pull/670/commits/eecaa80adf0b516a120cef03ee557712c163b0a4)? I recently made the change to merge the `ImageReader` logic (i.e. converting image to `blob`) into `ImageNormalizer` by creating the function `_convert_image_to_blob` in `ImageNormalizer`. Do I understand correctly that we also want to add the line (or anything that resembles the line) ` _blob = np.array(_blob, dtype='uint8')` in `ImageNormalizer`'s `_convert_image_to_blob`?

**Comment by nan-wang** [comment]
> Thank you for the suggestion. I was wondering if it's possible to just pass target_dtype into ImageNormalizer with numpy.uint8 as I believe this also works after the latest commit? I recently made the change to merge the ImageReader logic (i.e. converting image to blob) into ImageNormalizer by creating the function _convert_image_to_blob in ImageNormalizer. Do I understand correctly that we also want to add the line (or anything that resembles the line) _blob = np.array(_blob, dtype='uint8') in ImageNormalizer's _convert_image_to_blob?

@winstonww We don't need normalizer anymore. The `CLIPImageEncoder` can accept images of arbitrary sizes as long as it is `uint8`. Please refer more details to https://github.com/jina-ai/executor-clip-image#inputs

**Comment by winstonww** [comment]
As discussed with @numb3r3 , currently cross-modal-search example does not work well with docker. We suspect that the issue has to do with flow having both CLIPImageEncoder and CLIPTextEncoder in it, but this needs to be investigated. Nevertheless, the current cross-modal-search example runs successfully in local environment.

File: cross-modal-search/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM jinaai/jina:1.2.0
+FROM jinaai/jina:2.0.8-perf
```
Comment (winstonww) [inline cross-modal-search/Dockerfile:1]: @Yongxuanzhang  Version is now bumped to 2.0.8. However, we are seeing an issue happening when running cross-modal-search with docker, detailed [here](https://github.com/jina-ai/examples/pull/670#issuecomment-883190942). This needs to be further investigated. Thanks.

**Review by numb3r3 — APPROVED** [review]

**Pull request merged**